tests/aws/services/s3control/test_s3control.py

```diff
@@ -1,65 +1,421 @@
+            ),
+        ]
     )
 
-    assert put_response["ResponseMetadata"]["HTTPStatusCode"] == 200
 
-    get_response = s3control_client.get_public_access_block(AccountId=account_id)
-    assert access_block_config == get_response["PublicAccessBlockConfiguration"]
+@pytest.fixture
+def s3control_client(aws_client_factory, aws_client):
+    """
+    The endpoint for S3 Control looks like `http(s)://<account-id>.s3-control.<host>/v20180820/configuration/<path>
+    We need to manually set it to something else than `localhost` so that it is resolvable, as boto will prefix the host
+    with the account-id
+    """
+    if not is_aws_cloud():
+        return aws_client_factory(
+            aws_access_key_id=TEST_AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=TEST_AWS_SECRET_ACCESS_KEY,
+            endpoint_url=s3_control_endpoint,
+        ).s3control
+    else:
+        return aws_client.s3control
+
+
+@pytest.fixture
+def s3control_client_no_validation(aws_client_factory):
+    if not is_aws_cloud():
+        client = aws_client_factory(
+            config=Config(parameter_validation=False),
+            aws_access_key_id=TEST_AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=TEST_AWS_SECRET_ACCESS_KEY,
+            endpoint_url=s3_control_endpoint,
+        ).s3control
+    else:
+        client = aws_client_factory(config=Config(parameter_validation=False)).s3control
+
+    return client
+
+
+@pytest.fixture
+def s3control_create_access_point(s3control_client):
+    access_points = []
+
+    def _create_access_point(**kwargs):
+        resp = s3control_client.create_access_point(**kwargs)
+        access_points.append((kwargs["Name"], kwargs["AccountId"]))
+        return resp
+
+    yield _create_access_point
+
+    for access_point_name, req_account_id in access_points:
+        with contextlib.suppress(ClientError):
+            s3control_client.delete_access_point(AccountId=req_account_id, Name=access_point_name)
+
+
+class TestLegacyS3Control:
+    @markers.aws.validated
+    def test_lifecycle_public_access_block(self, s3control_client, account_id):
+        with pytest.raises(ClientError) as ce:
+            s3control_client.get_public_access_block(AccountId=account_id)
+        assert ce.value.response["Error"]["Code"] == "NoSuchPublicAccessBlockConfiguration"
+
+        access_block_config = {
+            "BlockPublicAcls": True,
+            "IgnorePublicAcls": True,
+            "BlockPublicPolicy": True,
+            "RestrictPublicBuckets": True,
+        }
+
+        try:
+            put_response = s3control_client.put_public_access_block(
+                AccountId=account_id, PublicAccessBlockConfiguration=access_block_config
+            )
+
+            assert put_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+            get_response = s3control_client.get_public_access_block(AccountId=account_id)
+            assert access_block_config == get_response["PublicAccessBlockConfiguration"]
```

Comment:
Oh no, this was a legacy test, I've just validated it but did not add snapshot. I think the goal would be to delete this one in favor of the newer ones created but currently skipped because the parity isn't there at all for now!

Reply:
:ok_hand:  makes total sense!